includes/admin/tools/class-sensei-tool-recalculate-course-enrolment.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * File containing Sensei_Tool_Recalculate_Course_Enrolment class.
+ *
+ * @package sensei-lms
+ * @since 3.7.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Sensei_Tool_Recalculate_Course_Enrolment class.
+ *
+ * @since 3.7.0
+ */
+class Sensei_Tool_Recalculate_Course_Enrolment implements Sensei_Tool_Interface {
+	const NONCE_ACTION = 'recalculate-course-enrolment';
+
+	/**
+	 * Get the ID of the tool.
+	 *
+	 * @return string
+	 */
+	public function get_id() {
+		return 'recalculate-course-enrolment';
+	}
+
+	/**
+	 * Get the name of the tool.
+	 *
+	 * @return string
+	 */
+	public function get_name() {
+		return __( 'Recalculate Course Enrollment', 'sensei-lms' );
+	}
+
+	/**
+	 * Get the description of the tool.
+	 *
+	 * @return string
+	 */
+	public function get_description() {
+		return __( 'Invalidate the cached enrollment and trigger recalculation for all users in a specific course.', 'sensei-lms' );
+	}
+
+	/**
+	 * Is the tool a single action?
+	 *
+	 * @return bool
+	 */
+	public function is_single_action() {
+		return false;
+	}
+
+	/**
+	 * Run the tool.
+	 */
+	public function run() {
+		// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Nonce checked in `process_input`.
+		if ( ! empty( $_POST['course_id'] ) ) {
+			$results = $this->process_input();
+
+			if ( $results ) {
+				wp_safe_redirect( Sensei_Tools::instance()->get_tools_url() );
```

Comment:
Since the `run()` is called from the `Sensei_Tools::output`, the headers issue also happens here.

Reply:
Should be fixed in v